flink-core/src/main/java/org/apache/flink/api/java/typeutils/ObjectArrayTypeInfo.java

```diff
@@ -72,15 +77,59 @@ public int getTotalFields() {
 
 	@Override
 	public boolean isKeyType() {
-		return false;
+		return true;
 	}
 
 	@SuppressWarnings("unchecked")
 	@Override
 	public TypeSerializer<T> createSerializer(ExecutionConfig executionConfig) {
 		return (TypeSerializer<T>) new GenericArraySerializer<C>(
-			componentInfo.getTypeClass(),
-			componentInfo.createSerializer(executionConfig));
+				componentInfo.getTypeClass(),
+				componentInfo.createSerializer(executionConfig));
+	}
+
+	@SuppressWarnings("unchecked")
+	private TypeComparator<? super Object> getBaseComparatorInfo(TypeInformation<? extends Object> componentInfo, boolean sortOrderAscending, ExecutionConfig executionConfig) {
```

Comment:
Why are you extracting for certain types the element comparator (for example the ObjectArrayTypeInfo) and for other you create the `TypeComparator` (for example the `CompositeTypeInfo`)? I don't get it. Why do you need the `getBaseComparatorInfo` method at all? Simply check in `createComparator` the different subtypes and then create the `TypeComparator`.